components/datetime/src/provider/timezones.rs

```diff
@@ -0,0 +1,123 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+use litemap::LiteMap;
+use std::borrow::Cow;
+
+/// Provides a few common map accessor methods to new-type structs that wrap a map.
+/// The methods are all pass-through calls to the internal methods of the same name.
+macro_rules! map_access {
+    ($outer: ty => $inner: ty: $lt: lifetime) => {
+        impl<$lt> $outer {
+            pub fn get<Q: ?Sized>(&self, key: &Q) -> Option<&$inner>
+            where
+                Q: Ord,
+                Cow<'s, str>: std::borrow::Borrow<Q>,
+            {
+                self.0.get(key)
+            }
+
+            pub fn is_empty(&self) -> bool {
+                self.0.is_empty()
+            }
+        }
+
+        impl<$lt, Q: ?Sized> std::ops::Index<&Q> for $outer
+        where
+            Q: Ord,
+            Cow<'s, str>: std::borrow::Borrow<Q>,
+        {
+            type Output = $inner;
+            fn index(&self, key: &Q) -> &Self::Output {
+                &self.0.get(key).unwrap()
+            }
+        }
+    };
+}
+
+/// An ICU4X mapping to the CLDR timeZoneNames format strings.
+/// See CLDR-JSON timeZoneNames.json for more context.
+#[derive(PartialEq, Debug, Clone)]
+#[cfg_attr(
+    feature = "provider_serde",
+    derive(serde::Serialize, serde::Deserialize)
+)]
+pub struct TimeZoneFormatsV1<'s> {
+    /// The hour format for displaying GMT offsets.
+    pub hour_format: Cow<'s, str>,
```

Comment:
`suggestion (if-minor)`: I know that for patterns we keep them currently stored unparsed, but this seems to be omnipresent (every hour_format has two elements) and easy to parse (it's always `X;Y`) - can we store it already as `hour_format: (Cow<'s, str>, Cow<'s, str>)` or with a semantic two element struct, since we will need to parse it before use?

Reply:
I like this idea. 

I think more optimizations will come along where, as we discussed, I can move more logic into the data provider side of things. 

For now, I just want to get an MVP provider/formatter working, even if it could use some improvements such as this. I appreciate you pointing these out.

Reply:
+1 on pre-parsing these patterns. I suggest doing that in a follow-up after #581 is merged.

Reply:
#581 will help with patterns, but the hour_format is `X;Y` and we don't need to block on #581.